pkgs/development/ruby-modules/gem-config/default.nix

```diff
@@ -135,6 +135,13 @@ in
     hardeningDisable = [ "format" ];
   };
 
+  rdiscount = attrs: {
+    # Disable randomized mangling to make outputs reproducible
+    dontBuild = false;
+    buildInputs = [ discount ];
```

Comment:
Is it not possible to build the gem with the vendored version and debian-glitch enabled? If the library author publishes a vendored version, it might cause some hair-pulling if the version doesn't match the API they are using. (this gem-config is used also by users outside of nixpkgs).

Not a blocker but I'm curious how much you looked into it.

Reply:
> Is it not possible to build the gem with the vendored version and debian-glitch enabled?

Sure, this works, although not by adding a flag only; requires a small patch, see #155363.

I thought it would be a good idea (and also a nixpkgs best practice?) to de-vendor the `discount` dependency to allow easy overriding which is properly propagated. I'm also fine with just patching the vendored source. Your call! 🙂

Reply:
I had a closer look and the discount library isn't changing a lot so it seems safe to un-vendor it. Also, the vendored library apparently has a CVE against it. https://github.com/davidfstr/rdiscount/issues/148

Reply:
This was also my assessment.

On a different note, is there any way to run the tests for Ruby packages? As far as I can tell, this isn't common.